### PR TITLE
Fix split bill flow in App

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
Refactors SplitBill to have the expected behavior

cc @mountiny @aldo-expensify since you reviewed the original split bill PR

### Fixed Issues
$ https://github.com/Expensify/App/issues/13378   

### Tests
**No extra groups created**
1. As memberA: create a groupDM with memberB, memberC & memberD
2. Go to the “+” menu inside the groupDM > click “Split bill”
3. Enter $100 > click Next
4. Deselect memberD > click Split $100
5. The split message from memberA is added to the groupDM: Split $100 with memberB and memberC
6. In the memberA:memberB DM the request action from memberA is added: Requested $33.33 from memberB
7. In the memberA:memberC DM the request action from memberA is added: Requested $33.33 from memberC
8. In the memberA:memberD DM there is nothing added, as memberD was excluded from the split.

**Splitting a bill with a single participant**
1. As memberA, create a groupDM with memberB and memberC
2. Go to the + menu inside the groupDM > click Split bill
3. Enter $100 > click Next
4. Deselect memberC > click Split $100
5. The split message from memberA is added to the groupDM: Split $100 with memberB
6. In the memberA:memberB DM the request action from memberA is added: Requested $50.00 from memberB
7. In the memberA:memberC DM there is nothing added, as memberC was excluded from the split.

**Split a bill from a policy room**
1. Create a Workspace `+ > New Workspace`
2. Navigate to `Workspace > Manage members > Invite`
3. Invite 10 members to the policy
4. In the `announce` room, select `Split bill` from the `+` icon inside the chat
5. Verify that there's a split message added to the room and no additional group chats are created

- [x] Verify that no errors appear in the JS console

### Offline tests
Same as Tests steps but the Split and Requested messages will be greyed out

### QA Steps
Same as test steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/22219519/206595276-a49cb7e5-3fc5-4149-a680-daf714ba4db4.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/22219519/206595315-76a1c12b-c04f-47af-a7e3-18c96f49d70c.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/22219519/206595335-143665ef-204e-4eac-bd29-a76574ef9ffb.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/22219519/206595351-3b4caa16-cc8a-460a-b27a-b6c03bc08b6c.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/22219519/206595383-4d81aba7-4285-474e-918b-d27e5e8c54e8.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/22219519/206595395-36b62c83-5674-4a7f-b41e-9bfb9c2d13a5.mov

</details>

### CODE REVIEW

**Comment by luacmartins** [comment]
@mananjadhav not yet! I’ll ping you once the API PR is merged!

File: src/libs/actions/IOU.js
Comment (mountiny) [inline src/libs/actions/IOU.js:339]:
```suggestion
        // If we only have one participant and the request was initiated from the global create menu, i.e. !existingGroupChatReportID, the oneOnOneChatReport is the groupChatReport
```

**Review by mountiny — COMMENTED** [review]
Just one NAB, changes look good to me otherwise

File: src/libs/actions/IOU.js
Comment (mountiny) [inline src/libs/actions/IOU.js:494]: @luacmartins As Monil pointed out, we are not passing the `existingGroupChatReportID` to the API here.

Comment (mountiny) [reply src/libs/actions/IOU.js:494]: requesting a change

**Review by mountiny — CHANGES_REQUESTED** [review]

File: src/libs/actions/IOU.js
Comment (luacmartins) [reply src/libs/actions/IOU.js:494]: We actually don't need to pass this param to the API. We have both `SplitBill` and `SplitBillAndOpenReport` commands and that's enough for the API to figure out what it needs to do without the need to pass this param.

Comment (luacmartins) [reply src/libs/actions/IOU.js:494]: The front end needs the ID to get the groupChat when it's created from within a chat though

**Comment by luacmartins** [comment]
I fixed the typo. Ready for another review!

File: src/libs/actions/IOU.js
Comment (mountiny) [reply src/libs/actions/IOU.js:494]: That makes sense!

**Review by mountiny — APPROVED** [review]

**Review by MonilBhavsar — COMMENTED** [review]
Code looks good. I'll get to testing today

**Comment by mananjadhav** [comment]
Quick bump is it ready for C+ review and testing?

**Comment by luacmartins** [comment]
@mananjadhav thanks for the bump. Yes, the API changes are in staging already!

**Comment by mananjadhav** [comment]
@luacmartins For the offline tests too it should post in the existing group?

Also I am currently testing this flow, but a bit blocked by the mobile web not working.

**Comment by mountiny** [comment]
> a bit blocked by the mobile web not working.

Is this because of the backend issues, right?

> For the offline tests too it should post in the existing group?

Yeah I would say the offline should do that same as online if successful

**Comment by mananjadhav** [comment]
Yeah none of us are able to run the app web or mobile web unless the IP is whitelisted

**Comment by mountiny** [comment]
I think it is ok at the current situation to skip the mweb or some internal employee can test those and you test the rest.

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/3069065/208481315-81cb7d78-ce84-41f5-a3b6-b20653597965.mov

https://user-images.githubusercontent.com/3069065/208481391-4007d39b-ac66-4662-9ac4-ce2dc5e8a00e.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/3069065/208483897-0a20cc5c-9504-40f3-badb-2a91b0d1cb3d.mov

https://user-images.githubusercontent.com/3069065/208484706-fb3e8189-6d95-4345-bb5e-dc1685247d87.mp4

https://user-images.githubusercontent.com/3069065/208484756-ba96bb28-ee1c-4f95-9ba2-8096b81a5584.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/3069065/208481528-2865f8ac-70dc-4537-85e9-25d828d4b14f.mp4

https://user-images.githubusercontent.com/3069065/208481597-fd2b2b91-4d9d-4428-aef7-3d40566f750f.mov

https://user-images.githubusercontent.com/3069065/208485036-36bc71f6-c07f-4716-ad49-992a6ee4909d.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/3069065/208482616-cde385e6-37a4-4bda-a89a-96c5280fba6a.mov

https://user-images.githubusercontent.com/3069065/208482732-4b96c1cf-c1dc-4aa1-a054-a214e36bc904.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/3069065/208482965-4a259f69-6f8a-460a-9f53-9562e1acc532.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/3069065/208483655-41eb9a8f-4ff6-412b-9f6e-9e9997aa31f4.mov

https://user-images.githubusercontent.com/3069065/208483855-5a1f2f02-72db-44ae-8f8d-7139d24e7e76.mov

</details>

---- 

@mountiny @MonilBhavsar I've tested the mobile web to the best possible means for me. I had a terrible network but it's IP was whitelisted so you can see a lot of lag there. But I managed to test it across all the platforms and have more than one videos for each test.

**Comment by luacmartins** [comment]
Thanks @mananjadhav. Still holding on Web-E.

**Comment by luacmartins** [comment]
Web-E is in prod. Merging this one!

**Pull request merged**
